libs/coin-modules/coin-sui/src/network/sdk.ts

```diff
@@ -230,33 +228,81 @@ export const executeTransactionBlock = async (params: ExecuteTransactionBlockPar
     return api.executeTransactionBlock(params);
   });
 
-// load from curos point or from begining until we reach the end
-const loadOperation = async (params: {
+/**
+ * Fetch operations for a specific address and type until the limit is reached
+ */
+export const loadOperations = async (params: {
   api: SuiClient;
-  type: OperationType;
   addr: string;
+  type: OperationType;
   cursor?: string | null | undefined;
 }): Promise<PaginatedTransactionResponse["data"]> => {
+  const operations: PaginatedTransactionResponse["data"] = [];
+  let currentCursor = params.cursor;
+
+  while (operations.length < TRANSACTIONS_LIMIT) {
+    try {
+      const { data, nextCursor, hasNextPage } = await queryTransactions({
+        ...params,
+        cursor: currentCursor,
+      });
+
+      operations.push(...data);
+
+      // If we got fewer results than the query limit or no more data, we've reached the end
+      if (data.length < TRANSACTIONS_LIMIT_PER_QUERY || !hasNextPage) {
+        break;
+      }
+
+      currentCursor = nextCursor;
+    } catch (error: unknown) {
+      if (
+        error !== null &&
+        typeof error === "object" &&
+        "type" in error &&
+        error.type === "InvalidParams"
```

Comment:
will `if (error?.type === "InvalidParams") { // ... }` be more cleaner?

Reply:
Sure, will fix

Reply:
Changed this